### ISSUE TITLE
Large kmem_alloc(55376, 0x1000)

### ISSUE THREAD

**Issue by cwtopel** [zfsonlinux/zfs#3945]
Issue starts at October 22 11:16:51 in the log file. 

http://hastebin.com/raw/akinafekat

**Comment by behlendorf** [comment]
Is this still an issue, and if so can you paste in an example back trace.

**Comment by cwtopel** [comment]
Looks like the paste expired, but basically the issue was that once I had exhausted my available RAM the system would attempt to flush to disk and it will lock up the entire system. 

My system is relatively old/slow (16GB DDR2, RAIDz1 with 2x 4TB 3GB/s SATA) and needed a little help moving things along.

[This](https://lonesysadmin.net/2013/12/22/better-linux-disk-caching-performance-vm-dirty_ratio/) is the workaround that I used: 

**/etc/sysctl.conf**
```
vm.dirty_background_ratio = 5
vm.dirty_ratio = 80
vm.swappiness = 1
```
I also switched all of my VMs over to `vm.swappiness = 1`

**Comment by kernelOfTruth** [comment]
@cwtopel what additional filesystems did you use ?

do you roughly still remember what was running back then ?

I'm asking because vm.dirty_background_ratio and vm.dirty_ratio don't have or shouldn't have an effect on ZFS

**Comment by cwtopel** [comment]
@kernelOfTruth The entire system was ZFS, no additional filesystems that I'm aware of. (The system was essentially running the Proxmox defaults.)

It's entirely possible that `vm.dirty_background_ratio` and `vm.dirty_ratio` aren't doing anything for me, as I added all of those options at roughly the same time. 

My running theory is that because a lot of my VMs were high-traffic they would swap internally and that would grind everything to a halt. So by limiting their ability to swap I avoided a lot of IO and the locking up of my storage pool. 

Vague details:
Proxmox 3.4/4.0, Debian Jessie
Kernels: 3.6.X & 4.X (This OOM error existed on both until I added the options to sysctl.conf)

That system has since been retired and I don't have access to any more information than that. :(
